Added a quick overview of the most used models. (#5284)
nautobot/nautobot: nautobot/docs/user-guide/core-data-model/overview/introduction.md

# Introduction

> Understanding the data model is key to effectively navigating Nautobot. The number of models in Nautobot Core alone is nearly 200 models which does not include any of the models that come from Nautobot apps. This introduction does not attempt to document all of models, but instead intends to highlight the most common models to be used as a quick reference.

Reviewer: ```suggestion
Understanding the data model is the key to effectively navigating Nautobot. There are nearly 200 models, which do not include any of the models that come from Nautobot apps, in Nautobot Core. This introduction does not attempt to document all of the models but instead intends to highlight the most common models to be used as a quick reference.
```

> ## Primary Model Simplified

Reviewer 2: Since `PrimaryModel` has a specific meaning within Nautobot, can we change this to something like `Basic Models` or `Fundamental Models` throughout this file?

> The models included are generally the most used models and the most used attributes of those models, with the following points of consideration:

Reviewer: ```suggestion
The models included are generally the most used models and attributes of those models, with the following points of consideration:
```

> - Tenants are an often used model, however they connect to most models and in the interest of simplicity not included.

Reviewer: I am not sure what this sentence is trying to say. Maybe?
```suggestion
- Tenants are an often used model. However, they connect to most models and are not included in the interest of simplicity.
```

Reviewer 2: I think the point here is that the Tenant model interconnects to most of these models but it's being intentionally omitted from the diagram to keep it more readable.

Author: @glennmatthews was correct, going to keep the original for now.

- The direction of relationship is not included, however you can generally tell if there is an `<model>_id` attribute, such as `device_id` on the Interface model.

> - Further details can be understood from the UML style diagram, that is roughly in the same layout.

Reviewer: ```suggestion
- More details can be found in the UML style diagram, which is roughly in the same layout.
```

> - This is a model view, not a table view, most noticeably that means "through" tables are not shown as they would in a UML diagram.

Reviewer: ```suggestion
- This is a model view and not a table view. So that means "through" tables are not shown as they would in a UML diagram.
```

![Fundamental Model Simplified](../../../media/models/model_simple.png)

## Fundamental Model UML

> The models included in the UML are largely the same as the simplified shown previously. This diagram was generated with postgres on [DBeaver](https://dbeaver.io/) and uses the [IDEF1X](https://en.wikipedia.org/wiki/IDEF1X) notation.

Reviewer: ```suggestion
The models included in the UML are largely the same as the simplified diagram shown previously. This diagram has been generated with Postgres on [DBeaver](https://dbeaver.io/) and uses the [IDEF1X](https://en.wikipedia.org/wiki/IDEF1X) notation.
```
